File: network/Mempool.cpp
//
// Created by Duarte on 26/10/2024.
//

#include "Mempool.h"
void Mempool::AddTransaction(const ITransaction& transaction) {
    transactions.push_back(transaction);
}

std::vector<ITransaction> Mempool::GetPendingTransactions() const {
    return transactions;
}

void Mempool::Clear() {
    transactions.clear();
}

File: network/mempool.ts
import type { IUTXO } from "./utxo.ts";

export interface IMempool{
	AddTransaction(transaction: IUTXO): void;
	GetPendingTransactions(): IUTXO[];
	Clear(): void;
}

export class Mempool implements IMempool{
  private transactions: IUTXO[] = [];

  public AddTransaction(transaction: IUTXO) {
    this.transactions.push(transaction);
  }

  public GetPendingTransactions(): IUTXO[] {
    return this.transactions;
  }

  public Clear() {
    this.transactions = [];
  }
}
